Development.md

```diff
@@ -0,0 +1,248 @@
+2. Set the `NEXT_PUBLIC_APP_ENV=development` in the .env
+3. Set the `MONGODB_URL=mongodb://127.0.0.1:27017/xai-app`
+    * note: if you have not "prepopulated" your local mongoDb, you need to do that next.
+    * For syncing testnet data ensure the current config file in packages/core/src/config.json is the Sepolia config file.
+    * Prepopulate mongoDb by starting the CLI and running the `sync-staking-pools` command.
+
+### Sentry Subgraph
+
+Sentry graph is deployed and hosted by alchemy https://subgraphs.alchemy.com/dashboard (Alchemy access required).
+
+The sepolia graphs are in `sentry-sepolia` https://subgraphs.alchemy.com/subgraphs/6221
+
+Build and deploy from the local machine in `infrastructure/sentry-subgraph` with `pnpm codegen`, `pnpm build` and the deployment cmd from the alchemy subgraph page. The sepolia sync has sepolia contract specific logic for managing Poolfactory and Referee versions as well as reducing the submissions entity count since sepolia has more challenges than one every hour during specific testing time frames. Sepolia deploys that sync should be done from `xai-foundation/sentry-development:subgraph-sepolia-sync` or a local branch off `xai-foundation/sentry-development:subgraph-sepolia-sync`. `xai-foundation/sentry-development:subgraph-sepolia-sync` needs to be kept up to date with sync script updates.
+
+For local testing the subgraph endpoint needs to be updated in `packages\sentry-subgraph-client\.graphclientrc.yml` and launched to a local playground by running `pnpm dev` in `packages\sentry-subgraph-client`.
+
+### Smart contracts
+
+All Xai ecosystem smart contracts are in `infrastructure/smart-contracts`, the deployments are managed through hardhat.
+
+Sepolia contracts can be deployed & upgraded with the sepolia deployer `0x490A5C858458567Bd58774C123250de11271f165` (Private Key required)
+The current sepolia infrastructure is deployed from `xai-foundation/sentry-development:sepolia-contract-deploy`, deployed hardhat config and .openzeppelin is at 
+- https://github.com/xai-foundation/sentry-development/tree/sepolia-contract-deploy/infrastructure/smart-contracts/sepolia.openzeppelin
+- https://github.com/xai-foundation/sentry-development/blob/sepolia-contract-deploy/infrastructure/smart-contracts/hardhat.config.cjs
+
+For deploying `sepolia.openzeppelin` needs to overwrite the existing `.openzeppelin`.
+
+All deployments and upgrades are managed by scripts in `infrastructure/smart-contracts/scripts`.
+
+#### Running the tests
+
+Update the `hardhat.config.cjs`, define the network to use in the `defaultNetwork` set to `hardhat`
+In `infrastructure/smart-contracts` run `pnpm local` for running a local node
+Run `pnpm test` to run all tests referenced in `infrastructure\smart-contracts\test\Fixture.mjs`
+
+---------------------------------
+
+## Production Environment
+
+### Operator & Web-Connect
+
+Push a **tag** on `xai-foundation/sentry:develop` to trigger the github actions. This will deploy
```

Comment:
Should this not be master for production?

Reply:
Yes thats right. However, for special cases we can push a tag from develop or even a working branch.
I updated the description.